stl/inc/mutex

```diff
@@ -508,61 +508,69 @@ public:
 };
 #endif // _HAS_CXX17
 
-// FUNCTION TEMPLATE _Invoke_stored_explicit
-template <class... _Types, size_t... _Indices>
-auto _Invoke_stored_explicit(tuple<_Types...>&& _Tuple, index_sequence<_Indices...>) -> decltype(
-    _STD invoke(_STD get<_Indices>(_STD move(_Tuple))...)) { // invoke() a tuple with explicit parameter ordering
-    return _STD invoke(_STD get<_Indices>(_STD move(_Tuple))...);
-}
-
-// FUNCTION TEMPLATE _Invoke_stored
-template <class... _Types>
-auto _Invoke_stored(tuple<_Types...>&& _Tuple)
-    -> decltype(_Invoke_stored_explicit(_STD move(_Tuple), index_sequence_for<_Types...>{})) { // invoke() a tuple
-    return _Invoke_stored_explicit(_STD move(_Tuple), index_sequence_for<_Types...>{});
-}
-
 // FUNCTION TEMPLATE call_once
-[[noreturn]] _CRTIMP2_PURE void __CLRCALL_PURE_OR_CDECL _XGetLastError();
-
-template <class _Tuple, class _Seq, size_t _Idx>
-int __stdcall _Callback_once(void*, void* _Pv, void**) { // adapt call_once() to callback API
-    _Tuple* _Ptup = static_cast<_Tuple*>(_Pv);
-
-    _TRY_BEGIN
-    // Note explicit _Seq{} selects every element from *_Ptup except the last,
-    // which contains call_once's exception_ptr.
-    _Invoke_stored_explicit(_STD move(*_Ptup), _Seq{});
-    _CATCH_ALL
-    auto& _Ref = _STD get<_Idx>(*_Ptup);
-    _Ref       = _STD current_exception();
-    return 0;
-    _CATCH_END
+#ifdef _M_CEE
+#define _WINDOWS_API              __stdcall
+#define _RENAME_WINDOWS_API(_Api) _Api##_clr
+#else // ^^^ _M_CEE // !_M_CEE vvv
+#define _WINDOWS_API              __declspec(dllimport) __stdcall
+#define _RENAME_WINDOWS_API(_Api) _Api
+#endif // _M_CEE
 
-    return 1;
-}
+// WINBASEAPI
+// BOOL
+// WINAPI
+// InitOnceBeginInitialize(
+//     _Inout_ LPINIT_ONCE lpInitOnce,
+//     _In_ DWORD dwFlags,
+//     _Out_ PBOOL fPending,
+//     _Outptr_opt_result_maybenull_ LPVOID* lpContext
+//     );
+extern "C" int _WINDOWS_API _RENAME_WINDOWS_API(__std_init_once_begin_initialize)(

```

Comment:
Questionable if it is correct that this function never fails on good parameters.

(But I'm still questioning that it does not fail on low resources. If it fails on low resources, then sure `_NODISCARD`)

Reply:
Either we don't need to test the return values at the callsites, in which case we shouldn't be testing the return values at the callsites, or we do need to test the return values at the callsites and we should annotate these with `_NODISCARD`.

Reply:
Can a check for preconditions-based error be done only under `#ifndef NDEBUG` ?

Reply:
The ice cold branch to abort is cheap insurance in case something bad happens here, so I'd do it unconditionally. The optimizer already knows to put any basic block that calls a noreturn function like abort out of the hot path.

Reply:
Then `_NODISCARD`?